docs/DeveloperGuide.md

```diff
@@ -249,47 +249,210 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
 
 
 Priority | As a ... | I want to ... | So that I can...
--------- | :-------- | :--------- | :-----------
+-------- | :---------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
-`* * *` | user | add a new person |
-`* * *` | user | delete a person | remove entries that I no longer need
-`* * *` | user | find a person by name | locate details of persons without having to go through the entire list
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
+`* * *` | user | add a new task (with deadlines) and events (with start and end timings) | keep track of it and complete it in the future
+`* * *` | user | delete a task | remove tasks that I no longer need to do
+`* * *` | user | edit a task | change the details of the tasks
+`* * *` | user | view tasks | decide on the follow-up action for each task
+`* * *` | user | clear all the data | remove all my information
+`* *` | user | prioritize my task | do the more important ones first
+`* *` | user | search for a task by keywords | view the details of task and complete it
+`* *` | user | undo a command | unexecute the last action that I just performed
+`* *` | user | add recurring tasks | save time entering the same task over multiple dates
+`* *` | user | choose my data storage location | have the flexibility to use the program on multiple computers as they can read from the same file stored on the cloud e.g. Google Drive
+`* *` | user | add a tag on tasks | categorize my task
+`* *` | user | edit a tag | rename the tag without the need to delete and add it again
+`* *` | user | mark my tasks as done or undone | change the status of my tasks
+`* *` | user | view tasks by tags/priority/date | group my tasks based on a field of my choice
```

Comment:
The reasoning part on "So that I can ......" is a little unclear, because the action doesn't exactly does any changes and groups the tasks to be saved or something. Do let me know if you understand what I mean ~ I think you can be more specific like viewing/displaying on certain type of tasks etc. 

Similar to previous comment, I suggest splitting this up but it is just a suggestion.